app/(general)/integration/livepeer/livestream/page.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import { useState } from 'react'
+
+import { LinkComponent } from '@/components/shared/link-component'
+import { FormLivepeerApiKey } from '@/integrations/livepeer/components/form-livepeer-api-key'
+import { useIsLivepeerApiKeySet } from '@/integrations/livepeer/hooks/use-livepeer-api-key'
+
+const newStreamObsPath = '/integration/livepeer/livestream/new/obs'
+const newStreamBrowserPath = '/integration/livepeer/livestream/new/browser'
+const watchStreamPath = '/integration/livepeer/livestream/watch'
+
+export default function PageIntegration() {
+  const [isLoadingNewStreamObs, setIsLoadingNewStreamObs] = useState<boolean>(false)
+  const [isLoadingNewStreamBrowser, setIsLoadingNewStreamBrowser] = useState<boolean>(false)
+  const [isLoadingExistingStream, setIsLoadingExistingStream] = useState<boolean>(false)
```

Comment:
It doesn't make sense to have a loading state on a client side navigation link. On next the Link href that are rendered are already prefetched and the target pages do not have any server side rendered content, so the page will always load instantly. Remember to never design an app based on the behaviour of the development server and always consider the production behaviour 😉

Reply:
Also, it doesn't make much sense to wrap a button into a link. Links are for navigation and buttons for actions. You should use one or other. In cases where you want to also use a callback in a navigation you can use the `onClick` property of Link or the anchor tag

Reply:
thanks!

Reply:
Agree with you on the second point but all the integrations use this structure e.g. [Disco Integration](https://github.com/turbo-eth/template-web3-app/blob/ac480565b67ae1b3f29cc5399291d4c104b8a323/app/(general)/integration/disco/page.tsx#L55C13-L55C13) and [Lit Protocol](https://github.com/turbo-eth/template-web3-app/blob/ac480565b67ae1b3f29cc5399291d4c104b8a323/app/(general)/integration/lit-protocol/layout.tsx#L53)

Should I open a PR later to solve all of them at once?

Reply:
oh thanks for pointing that I've completely missed it. It would be great if you could open a PR fixing it :D

Reply:
Done! https://github.com/turbo-eth/template-web3-app/pull/117